software/Makefile

```diff
@@ -16,17 +16,19 @@ SED ?= sed
 # 120 MB - 4096
 TOP_OF_MEM_RISC_V=0x877FF000
 
-FILTER_OUT = $(foreach v,$(2),$(if $(findstring $(1),$(v)),,$(v)))
+# test command avoids error if $(SDK_DIR)/prog does not yet exist
+all-riscv-binaries:=$(shell test -d $(SDK_DIR)/prog && find $(SDK_DIR)/prog -iname "*.riscv")
+riscv-dont-run=arch.mk Config.in
+$(warning filtering $(riscv-dont-run))
+riscv-binaries=$(filter-out $(addprefix %,$(addsuffix .riscv,$(riscv-dont-run))),$(all-riscv-binaries))
+$(warning $(riscv-binaries))
 
-riscv-binaries:=$(shell find ../. -iname "*.riscv")
 all-nbf-binaries:=$(shell find . -iname "*.nbf")
 nbf-dont-run = bootrom 181.mcf 454.calculix 458.sjeng 471.omnetpp
 $(warning filtering $(nbf-dont-run))
-
-nbf-files=$(call FILTER_OUT,$(nbf-dont-run),$(all-nbf-binaries))
+nbf-files=$(filter-out $(addprefix %,$(addsuffix .nbf,$(nbf-dont-run))),$(all-nbf-binaries))
```

Comment:
What’s the reason for the change?

Reply:
the FILTER-OUT function doesn't work based on my testing. I think it requires an exact match of the full filepath, so the "dont-run" things would need to be full filepaths. This way we add a wildcard at the front of each -dont-run (e.g., %bootrom.nbf), and then the built-in filter-out works as expected.

Reply:
@farzamgl has this target worked for you?